docs/refs/api/rtos/callbacks.md

```diff
@@ -169,25 +70,13 @@ int main() {
 }
 ```
 
-#### Combinatorial explosion damage control (aka why a separate class for Callbacks)
+#### A separate class for Callbacks)
 
-Unfortunately, supporting all of the standard C++ function types is difficult.
+C++ requires a large set of overloads to support all of the standard function types. It is unreasonable to expect a new library author to add all of these overloads to every function that could take in a callback.
 
-1. State is important, so need to support either C-style function pointers with state, or C++ member function pointers.
+However, C++ provides the tools to delegate this problem to a single class. This class is the Callback class. The Callback class should be familiar to users of the `std::function` class that C++11 introduced but is available for older versions of C++.
 
-2. Stateless callbacks are just as common, but passing a stateless callback as a member function function requires writing a lot of boilerplate code and instantiating an empty class. So we need to also support a standard function pointer.
-
-3. Another design pattern you may see is the function object, a class that overrides the function call operator. We can expect the user to pass function objects as C++ member function pointers if needed.
-
-4. A useful C++ feature is the enforcement of const-correctness, but this becomes unfortunately complicated with the state associated with callbacks. A C++ API needs to support both the const and non-const versions of member function pointers.
-
-5. Another C++ feature is volatile-correctness in case the underlying state must be volatile, but if necessary we can probably expect the user to hide volatile members inside of a non-volatile class.
-
-Long story short, C++ requires a large set of overloads to support all of the standard function types. It is very error-prone and unreasonable to expect a new library author to add all of these overloads to every function that could take in a callback.
-
-Fortunately, C++ does provide the tools to delegate this problem to a single class. This class is the Callback class. The Callback class should be familiar to users of the std::function class that was introduced in C++11, but is available for older versions of C++.
-
-**An overly-simplified description of the Callback class is that is contains all of this madness so you don’t have to.**
+**An overly-simplified description of the Callback class is that it contains all of this madness, so you don’t have to.**
```

Comment:
The "this madness" in the sentence doesn't refer to anything if we drop the previous bullet points. Are we just wanting to drop this section?